### PR TITLE
Fix date issue by switching to converting UV data using location-based timezones

### PR DESCRIPTION
**Pull request by lindsayplatt**
Finally committing/PR-ing the code related to my changes to summarize UV data based on timezone, so that we don't end up with data from the wrong day getting averaged together. The reason we noticed this in the first place was because we kept ending up with an extra frame outside of the timeframe we specified (see #47).

### CODE REVIEW

**Comment by lindsayplatt** [comment]
See https://github.com/USGS-R/dataRetrieval/issues/634#issuecomment-1249735372 for how I figured out the daylight savings issue fix.

**Comment by lindsayplatt** [comment]
I've verified that this works and we no longer have dates that extend beyond the dates we requested. So, each summarized daily value is now local to each site's timezone.

```r
library(dplyr)
readr::read_csv('2_process/out/gw_daily_quantiles.csv') %>%
  mutate(day_seq = lubridate::yday(Date)) %>%
  group_by(day_seq) %>%
  summarize(sites = length(unique(site_no))) %>%
  arrange(desc(day_seq)) 

# A tibble: 365 x 2
   day_seq sites
     <dbl> <int>
 1     365  1823
 2     364  1819
 3     363  1817
 4     362  1820
 5     361  1818
 6     360  1817
 7     359  1822
 8     358  1821
 9     357  1825
10     356  1833
```

**Comment by lindsayplatt** [comment]
@ceenell requesting a review but I don't expect you to run this. I have re-run all this code locally and tested the new function I created via https://github.com/USGS-R/dataRetrieval/issues/634#issuecomment-1249735372. Just hoping to have your eyes scan this before it officially is merged.

File: 1_fetch/src/fetch_nwis.R [hunk 1]
```diff
@@ -30,14 +30,110 @@ fetch_gw_uv <- function(target_name, gw_sites, start_date, end_date, param_cd) {
...
+    
+    ### Reduce each instantaneous value to a single average for each date
     group_by(site_no, Date) %>% 
     summarize(GWL = mean(GWL_inst, na.rm = TRUE)) %>% 
     write_feather(target_name)
 }
 
+# Convert POSIXct to Dates for a given timezone
+# Only works with one tz_abbr at a time
+POSIXct_to_Date_tz <- function(posix_dates, tz_abbr) {
```
Comment (ceenell) [inline 1_fetch/src/fetch_nwis.R:62]: These next two function are duplicated in `0_historic/src/fetch_nwis_historic`

Comment (ceenell) [reply 1_fetch/src/fetch_nwis.R:62]: I'm not sure I understand the need for doing that. I know we're keeping the historic fetch separate from the current, but seems like duplicating the same steps increases complexity and room for error

File: 1_fetch/src/fetch_nwis.R [hunk 2]
```diff
@@ -30,14 +30,110 @@ fetch_gw_uv <- function(target_name, gw_sites, start_date, end_date, param_cd) {
     write_feather(target_name)
 }
 
-convert_uv_to_dv <- function(target_name, gw_uv_data_fn) {
+convert_uv_to_dv <- function(target_name, gw_uv_data_fn, site_tz_xwalk) {
```
Comment (ceenell) [inline 1_fetch/src/fetch_nwis.R:33]: Is this different from `fetch_nwis_historic.r`?

**Review by ceenell — COMMENTED** [review]
That's great this mystery is is resolved and just wow.  How many sites was this an issue for? Is this something that also needs to be addressed with `gage-conditions-gif`?

Without running the pipeline I cannot confirm that this works, but nothing stands out to me as an error with the syntax. 

I am wondering about the duplicated functions between historic and current data pulls. It is not immediately clear to me why it's needed.

File: 1_fetch/src/fetch_nwis.R (hunk 2 above)
Comment (lindsayplatt) [reply 1_fetch/src/fetch_nwis.R:33]: Nope ... will address this in #74

**Review by ceenell — APPROVED** [review]
OK go ahead and merge to keep things moving forward

**Comment by lindsayplatt** [comment]
As for how many sites this timezone problem was impacting, I believe based on your findings [here](https://github.com/USGS-VIZLAB/gw-conditions/issues/47#issuecomment-1008514542), it was almost 600 out of the ~1300 total sites.

**Comment by lindsayplatt** [comment]
Lastly, you asked if this is an issue for gage-conditions-gif. My guess is yes, but we haven't noticed because in `gage-conditions-gif`, we [set the PNG frame dates based on the supplied viz dates](https://github.com/USGS-VIZLAB/gage-conditions-gif/blob/main/2_process.yml#L51-L52), rather than [using the data to generate the SVG visuals](https://github.com/USGS-VIZLAB/gw-conditions/blob/main/3_visualize/src/svg_utils_vue.R#L2) as we do in this repo.

**Pull request merged**